### PR TITLE
Improving status reporting

### PR DESCRIPTION
**Pull request by dschuesae**
Refactoring and moved it to better places. I hope this is not too much reporting. But as HASS has this msgRateSensor it should not be the case.

### CODE REVIEW

File: lib/TWCManager/TWCSlave.py [hunk 1]
```diff
@@ -461,11 +462,24 @@ def receive_slave_heartbeat(self, heartbeatData):
         self.reportedAmpsActual = ((heartbeatData[3] << 8) + heartbeatData[4]) / 100
         self.reportedState = heartbeatData[0]
 
+        if (self.reportedAmpsActual != self.reportedAmpsLast):
```
Comment (MikeBishop) [inline lib/TWCManager/TWCSlave.py:465]: By moving it here, you're bypassing the "significant change" logic. Is that intentional? If so, why?

Comment (dschuesae) [reply lib/TWCManager/TWCSlave.py:465]: I agree. To be honest - I saw this a week ago, but it did not work multiple times. I had always a value there for more the whole night, even without charging. By the way in the HASSStatus class there is already a limitation for too much updating. I will have another look here...

**Comment by dschuesae** [comment]
Just added sending creating a message queue for status reporting to HASS. Should update the sensors every 15 seconds if changed or every hour if no change occured. So the sensor will still be valid even it is not changing a long long time.

File: lib/TWCManager/TWCSlave.py (hunk 1 above)
Comment (dschuesae) [reply lib/TWCManager/TWCSlave.py:465]: In my opinion it should exactly be updated there and not afterwards. The overload of status reporting should be handled in the Status classes. To be clear - I had it several times during night - that charging was over and it still shows me power... Now it works like a charm

**Comment by dschuesae** [comment]
So in my opinion this should now be completely backward comaptible.

**Comment by MikeBishop** [comment]
I gave this a read through, and it looks reasonable.  It appears that the MQTT module already has rate limiting, so it should be okay with the faster updates.

**Pull request merged**